authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -307,16 +319,24 @@ getAccessTokenProxy, getTokenCredentialProxy
                -> m Credential -- ^ Token Credential (Access Token & Secret)
 getAccessTokenProxy p = getAccessToken' $ addMaybeProxy p
 
-
 getAccessToken' :: MonadIO m
                 => (Request -> Request)       -- ^ Request Hook
                 -> OAuth                      -- ^ OAuth Application
                 -> Credential                 -- ^ Temporary Credential (with oauth_verifier if >= 1.0a)
                 -> Manager
                 -> m Credential     -- ^ Token Credential (Access Token & Secret)
-getAccessToken' hook oa cr manager = do
+getAccessToken' = getAccessToken'' addAuthHeader
+
+getAccessToken'' :: MonadIO m
```

Comment:
When I run into a situation like this, I generally like to abstract the arguments out into a separate datatype so that new fields can be added in the future without breaking backwards compatibility. It might be time to do that here. The standard naming convention for such a situation would then be `getAccessTokenWith`.